app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,54 @@
+class PostsController < ApplicationController
+  
+  before_action :authenticate_user!
```

Comment:
We need this for all the other controllers. Please think of how to incorporate this into the others